pkg/ruleengine/v1/r0006_unexpected_service_account_token_access.go

```diff
@@ -42,15 +41,68 @@ var R0006UnexpectedServiceAccountTokenAccessRuleDescriptor = ruleengine.RuleDesc
 		return CreateRuleR0006UnexpectedServiceAccountTokenAccess()
 	},
 }
-var _ ruleengine.RuleEvaluator = (*R0006UnexpectedServiceAccountTokenAccess)(nil)
 
 type R0006UnexpectedServiceAccountTokenAccess struct {
 	BaseRule
 }
 
+// getTokenBasePath returns the base service account token path if the path is a token path,
+// otherwise returns an empty string. Using a single iteration through prefixes.
+func getTokenBasePath(path string) string {
+	for _, prefix := range serviceAccountTokenPathsPrefixes {
+		if strings.HasPrefix(path, prefix) {
+			return prefix
+		}
+	}
+	return ""
+}
+
+// normalizeTokenPath removes timestamp directories from the path while maintaining
+// the essential structure. Optimized for minimal allocations.
+func normalizeTokenPath(path string) string {
```

Comment:
do you have a time constraint on this? because splitting and joining strings are really wasteful
you can check how much fun we had with indexing strings in https://github.com/kubescape/storage/blob/2b75b33130a55255df7575dc473a49858e3e325d/pkg/registry/file/dynamicpathdetector/analyzer.go#L137